reader.c

```diff
@@ -151,6 +212,10 @@ int nsq_reader_connect_to_nsqd(struct NSQReader *rdr, const char *address, int p
     struct NSQDConnection *conn;
     int rc;
 
+    if (rdr->lookupd == NULL) {
+        rdr->address = strdup(address);
+        rdr->port = port;
+    }
```

Comment:
@mreiferson I see it now. I am going to fix that. BTW, is the title of this PR still appropriate? 😄